packages/flutter/test/cupertino/dialog_test.dart

```diff
@@ -103,128 +89,103 @@ void main() {
-        child: new Center(
-          child: new Builder(builder: (BuildContext context) {
-            return new RaisedButton(
-              onPressed: () {
-                showDialog<void>(
-                  context: context,
-                  builder: (BuildContext context) {
-                    return new MediaQuery(
-                      data: MediaQuery.of(context).copyWith(textScaleFactor: 3.0),
-                      child: new CupertinoAlertDialog(
-                        title: const Text('The Title'),
-                        content: new Text('Very long content ' * 20),
-                        actions: const <Widget>[
-                          CupertinoDialogAction(
-                            child: Text('Cancel'),
-                          ),
-                          CupertinoDialogAction(
-                            isDestructiveAction: true,
-                            child: Text('OK'),
-                          ),
-                        ],
-                        scrollController: scrollController,
-                      ),
-                    );
-                  },
-                );
-              },
-              child: const Text('Go'),
-            );
-          }),
-        ),
-      )),
+      createAppWithButtonThatLaunchesDialog(
+        dialogBuilder: (BuildContext context) {
+          return new MediaQuery(
+            data: MediaQuery.of(context).copyWith(textScaleFactor: 3.0),
+            child: new CupertinoAlertDialog(
+              title: const Text('The Title'),
+              content: new Text('Very long content ' * 20),
+              actions: const <Widget>[
+                CupertinoDialogAction(
+                  child: Text('Cancel'),
+                ),
+                CupertinoDialogAction(
+                  isDestructiveAction: true,
+                  child: Text('OK'),
+                ),
+              ],
+              scrollController: scrollController,
+            ),
+          );
+        }
+      )
     );
 
     await tester.tap(find.text('Go'));
-
     await tester.pump();
-    await tester.pump(const Duration(seconds: 1));
 
     expect(scrollController.offset, 0.0);
     scrollController.jumpTo(100.0);
     expect(scrollController.offset, 100.0);
     // Set the scroll position back to zero.
     scrollController.jumpTo(0.0);
 
-    // Find the actual dialog box. The first decorated box is the popup barrier.
-    expect(tester.getSize(find.byType(DecoratedBox).at(1)), equals(const Size(270.0, 560.0)));
+    // Expect the modal dialog box to take all available height.
+    expect(
+      tester.getSize(
+        find.byType(ClipRRect)
+      ),
+      equals(const Size(270.0, 560.0)),
+    );
 
-    // Check sizes/locations of the text.
+    // Check sizes/locations of the text. The text is large so these 2 buttons are stacked.
+    // Visually the "Cancel" button and "OK" button are the same height. I don't know why
```

Comment:
You should figure out the answer to the question in the comment here before checking it in. Don't check in code you don't understand.

Reply:
@Hixie - How much time is it worth?  This issue was already present.  Greg doesn't remember why it was the case either.  I looked into Xiao's theory about the GestureDetector but that wasn't it.  I've rendered with layout boundaries and they look fine.  Is this issue worth hours? days? weeks?

Reply:
Several days, certainly, maybe a week or two. This looks like a serious problem, who knows what else it could be causing.

It shouldn't take long to figure out, though. Start with the test that's misbehaving, and start simplifying, substituting in composition, moving classes to the test file and simplifying them down, etc, until you're down to a very small test that reproduces the problem, which uses nothing but the buggy code. I'm happy to help if you want.